src/shared/watchdog.c

```diff
@@ -95,6 +131,34 @@ static int watchdog_ping_now(void) {
         return 0;
 }
 
+static int update_pretimeout(void) {
+        int r, t_sec, pt_sec;
+
+        if (watchdog_fd < 0)
+                return 0;
+
+        if (watchdog_pretimeout == USEC_INFINITY)
+                return 0;
+
+        /* Determine if the pretimeout is valid for the current watchdog timeout. */
+        t_sec = saturated_usec_to_sec(watchdog_timeout);
```

Comment:
this is not going to do anything useful in case watchdog_timeout == USEC_INFINITY, no?

Reply:
I'm not sure if it's allowed to enable a pretimeout without a timeout - if it is, this will do that

Reply:
oh, i am not saying it would. just saying this needs a guard in some way, here. i.e.e we should refuse that here.